nvflare/app_common/pt/pt_file_model_locator.py

```diff
@@ -67,7 +68,10 @@ def locate_model(self, model_name, fl_ctx: FLContext) -> DXO:
         if model_name not in list(self.model_inventory.keys()):
             raise ValueError(f"model inventory does not contain: {model_name}")
 
-        weights = self.model_persistor.get_model(model_name, fl_ctx)
+        model_learnable = self.model_persistor.get_model(model_name, fl_ctx)
+        weights = model_learnable.get(ModelLearnableKey.WEIGHTS)
+        if weights is None:
+            raise ValueError(f"model {model_name} does not have required model weights.")
         dxo = DXO(data_kind=DataKind.WEIGHTS, data=weights, meta={})
```

Comment:
Should change to use model_learnable_to_dxo(model_learnable) here. It also includes all meta, which will keep the DXO self-sufficient.

Reply:
Updaed to use model_learnable_to_dxo().